### PR TITLE
Add ability to export attributes corresponding to `Logrecord.EventId`

### PR DESCRIPTION
**Pull request by vishweshbankwar**
towards #4875
Design discussion issue #

## Changes

Introduces `OTEL_DOTNET_EXPERIMENTAL_OTLP_EMIT_EVENT_LOG_ATTRIBUTES` for exporting corresponding to `LogRecord.EventId`. The attributes will be exported as `logrecord.event.id` and `logrecord.event.name`.

## Merge requirement checklist

* [ ] [CONTRIBUTING](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/CONTRIBUTING.md) guidelines followed (nullable enabled, static analysis, etc.)
* [x] Unit tests added/updated
* [x] Appropriate `CHANGELOG.md` files updated for non-trivial changes
* [ ] Changes in public API reviewed (if applicable)

### CODE REVIEW

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/README.md
Comment (rajkumar-rangaraj) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/README.md:234]: nit: consider - When set to true, it exports LogRecord.EventId as event.id, LogRecord.CategoryName as dotnet.ilogger.category, and event.name.

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/CHANGELOG.md
Comment (rajkumar-rangaraj) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/CHANGELOG.md:22]:
```suggestion
* Added ability to export attributes corresponding to `LogRecord.EventId` as
`event.id` , `LogRecord.CategoryName` as
`dotnet.ilogger.category` and `event.name` . These attributes will be exported when
`OTEL_DOTNET_EXPERIMENTAL_OTLP_EMIT_CATEGORY_AND_EVENT_LOG_ATTRIBUTES` will be
set to `true`.
```

**Review by rajkumar-rangaraj — APPROVED** [review]
LGTM

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/README.md
Comment (utpilla) [reply src/OpenTelemetry.Exporter.OpenTelemetryProtocol/README.md:234]: It would export `LogRecord.EventId.Id` as `event.id` and `LogRecord.EventId.Name` as `event.name`.

**Review by utpilla — APPROVED** [review]

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/OtlpLogRecordTransformer.cs
Comment (cijothomas) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/OtlpLogRecordTransformer.cs:108]: "dotnet.ilogger.event.id" have we considered this, given we prefix "dotnet.ilogger" to the category name?

**Comment by cijothomas** [comment]
resolves https://github.com/open-telemetry/opentelemetry-dotnet/issues/4404 too

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ExperimentalOptions.cs
Comment (utpilla) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ExperimentalOptions.cs:28]: It might be a good idea to have dedicated environment variables for EventId and CategoryName. If one ofthem gets stabilized before the other, it would be odd to use this name `EMIT_CATEGORY_AND_EVENT_LOG_ATTRIBUTES` for the other one.

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/CHANGELOG.md
Comment (cijothomas) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/CHANGELOG.md:27]: I think its good to mention the full context here for people upgrading from pre-stable, where we used to export this as a diff. attribute.
Something like. versions before A used to export this as attribute ...
version B removed this.
version C (current), adds this under feature flag,. Note that the attribute names are changed as well.

(So end users will have easier time following the changelog, instead of having to open multiple PR links and understand what changed if upgrdaing from 1.6.0.beta to 1.7.0.beta..

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/OtlpLogRecordTransformer.cs
Comment (cijothomas) [inline src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/OtlpLogRecordTransformer.cs:108]: @ThomsonTan Could you update how OTel Cpp exports its eventid, given OTel Cpp is the only other language with eventid being a first class thing?

Comment (ThomsonTan) [reply src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/OtlpLogRecordTransformer.cs:108]: EventId is currently ignored when exporting to OTLP in OTel Cpp. I thought we should create a semantic convention for it to make the implementation across languages consistent? And don't add any language specific name to the attribute name.

Comment (cijothomas) [reply src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/OtlpLogRecordTransformer.cs:108]: Yes that makes sense. Given this is experimental, it'd be okay to change the attribute name from event.id to something else (if the spec/sem. convention warrants it).

@vishweshbankwar Could you open a spec issue to create a semantic convention for event.id (a numerical identifier to be attached to individual logs), and add a comment linking to that?

**Review by cijothomas — APPROVED** [review]
Looks good to ship this under experimental flag, given there is no existing semantic convention for this. 

Would be nice to make the changelog more self-contained to help end users navigate these changes more smoothly.

**Comment by cijothomas** [comment]
> resolves #4404 too

@nblumhardt Could you review as well?

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ExperimentalOptions.cs
Comment (vishweshbankwar) [reply src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/ExperimentalOptions.cs:28]: I am good with further categorizing these. I will make the change.

File: src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/OtlpLogRecordTransformer.cs
Comment (vishweshbankwar) [reply src/OpenTelemetry.Exporter.OpenTelemetryProtocol/Implementation/OtlpLogRecordTransformer.cs:108]: Category name is still open for discussion. https://github.com/open-telemetry/opentelemetry-dotnet/issues/3491.

I do not have strong take on this yet. If we decide to use `InstrumentationScope` for category then do we need still need this?

Something to be discussed on the issue. Comments are misleading in code (I will fix it).

**Comment by CodeBlanch** [comment]
I think we should shy away from using `event.id` & `event.name` for the keys. Reason being, there is already some definition applied to those: https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/events.md#general-event-attributes

`event.name` is _supposed_ to go with `event.domain`

Why tie ourselves to that `event` stuff though? Seems like some other name would work just fine. `dotnet.logrecord.event.id` + `dotnet.logrecord.event.name` or whatever.

/cc @alanwest

**Comment by alanwest** [comment]
> `event.name` is _supposed_ to go with `event.domain`

Agreed. @vishweshbankwar and I spoke about this yesterday. If we are adopting the event conventions then I think it is inappropriate to only adopt them halfway.

@vishweshbankwar you were going to discuss this with other. Any updates? Any ideas of what to set `event.domain` to?

The idea I shared was `event.domain=ilogger`

> Why tie ourselves to that event stuff though?

I'm still not a fan of adopting them, however, as I discussed with @vishweshbankwar, I can get behind it if we embrace the spirit of applying the experimental conventions to help drive them forward based on feedback we may learn. What this looks like to me is:

1) We need to embrace the conventions as they currently are - i.e., set `event.domain` and `event.name`
2) If we're going to introduce `event.id` then someone needs to champion moving this into the spec.

**Review by CodeBlanch — APPROVED** [review]
LGTM

Going to try and push clarification through the spec on this: https://github.com/open-telemetry/semantic-conventions/issues/372

**Review by nblumhardt — COMMENTED** [review]
In .NET, `EventId.Id` and `EventId.Name` are generally only unique/useful in the context of an `ILogger` category:

> Event IDs and names should be unique so that different message types can be identified.  ([source](https://learn.microsoft.com/en-us/dotnet/core/compatibility/aspnet-core/7.0/microsoft-aspnetcore-mvc-core-log-event-ids))

So if they were to be translated into `event.id` and `event.name` (seems fine to me) then perhaps the category belongs alongside them in an `event.*` attribute, too?

Putting the category into `event.domain` instead of `dotnet.ilogger.category` would make this intent clear and qualify the names/ids nicely:

```
event.id = 7
event.name = UserCreated
event.domain = Microsoft.AspNetCore.Identity.UI
```

Unfortunately the definition of `event.domain` seems to exclude this usage, but, perhaps another `event.*` attribute can be coined? `dotnet.ilogger.category` is a bit of an oddity right now and seems like a good candidate for unification into something more general.

**Comment by utpilla** [comment]
> Any ideas of what to set event.domain to?

@alanwest Firstly, I think deciding this would take some more time and discussion and we need not block this PR for that. We don't have any users asking us to export `event.domain` attribute so this discussion can be continued even after this PR gets merged.

Now if we could go with Proposal A in https://github.com/open-telemetry/semantic-conventions/issues/372, then we wouldn't have to worry about `event.domain`.

In the case where we have to go forward with Proposal B/ Event semantic conventions, I feel that `CategoryName` would be a good candidate for `event.domain`.

The current description for `event.domain` on the [spec](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/events.md#:~:text=The%20domain%20identifies%20the%20business%20context%20for%20the%20events.%20%5B1%5D) says that:
> The domain identifies the business context for the events. [1]
> [1]: Events across different domains may have same event.name, yet be unrelated events.

`CategoryName` fits that description well. For example, an ASP .NET Core app could have multiple loggers each with the category name matching the controller name. That offers a logical grouping of the logs emitted by a given controller. Within each controller, we could have defined events and different controllers could potentially have events with the same name.

**Comment by alanwest** [comment]
> CategoryName fits that description well.

We now have two people on this PR who have expressed this intuition. It's also an option @CodeBlanch suggested long ago. Is there a reason we shouldn't set `event.domain` to `CategoryName` now?

**Comment by vishweshbankwar** [comment]
> > CategoryName fits that description well.
> 
> We now have two people on this PR who have expressed this intuition. It's also an option @CodeBlanch suggested long ago. Is there a reason we shouldn't set `event.domain` to `CategoryName` now?


Let me take a step back and summarize what I think we should do:

### P0 - Resolve issues reported by users

#### 1) Issue of missing attributes

These attributes have been part of the exporter since the very beginning, and not having the ability to get these attributes is disruptive for users.

- See:
  - [Discussion](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4781#discussion_r1316251649)
  - [Issue](https://github.com/open-telemetry/opentelemetry-dotnet/issues/4831)

Additionally, users have asked about this offline as well.

#### 2) Issue of name and id being too generic

[Issue](https://github.com/open-telemetry/opentelemetry-dotnet/issues/4404)

This PR **addresses** both of these feedbacks.

### P1 - Stabilize these experimental attributes

All the discussions here on this PR are a good step forward towards that.

There is already some engagement on the issue opened by [@CodeBlanch](https://github.com/open-telemetry/semantic-conventions/issues/372).

So, in favor of **addressing P0**, I think we should go ahead with this PR. Additionally, this will help with getting more feedback as well.

We should continue our discussions on **P1** on [semantic-conventions issue](https://github.com/open-telemetry/semantic-conventions/issues/372) and [opentelemetry-dotnet issue](https://github.com/open-telemetry/opentelemetry-dotnet/issues/3491).

For the part of setting the category to `event.domain`, Here is what we need to consider further:

1) The description in the [table](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/events.md#:~:text=The%20domain%20identifies%20the%20business%20context%20for%20the%20events.%20%5B1%5D) looks a bit confusing after looking at the example [values](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/events.md#:~:text=event.domain%20has,Events%20from%20Kubernetes) and based on another [statement](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/general/events.md#:~:text=The%20event.domain,of%20the%20domains). This does not match with what the category name [suggests](https://learn.microsoft.com/en-us/dotnet/core/extensions/logging?tabs=command-line#log-category:~:text=The%20category%20string%20is%20arbitrary%2C%20but%20the%20convention%20is%20to%20use%20the%20class%20name.%20For%20example%2C%20in%20an%20application%20with%20a%20service%20defined%20like%20the%20following%20object%2C%20the%20category%20might%20be%20%22Example.DefaultService%22)

2) [OpenTelemetry Specification Issue](https://github.com/open-telemetry/opentelemetry-specification/issues/2994).

3) Should this be [InstrumentationScope](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#field-instrumentationscope) `name` field? The definition matches closely with the Category Name.

That being said, I agree that **P1** still remains unaddressed but it should be handled separately from this PR.

**Comment by nblumhardt** [comment]
> Should this be [InstrumentationScope](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/logs/data-model.md#field-instrumentationscope) name field? The definition matches closely with the Category Name.

Using instrumentation scopes could be suboptimal for libraries/the SDK generating OTLP payloads, since (as far as I understand it) "log records" nest under "scope logs" and so the SDK would need to sort/group events in order to build properly nested/non-degenerate OTLP payloads.

**Comment by reyang** [comment]
> > CategoryName fits that description well.
> 
> We now have two people on this PR who have expressed this intuition. It's also an option @CodeBlanch suggested long ago. Is there a reason we shouldn't set `event.domain` to `CategoryName` now?

+1, `event.domain` is better than `dotnet.ilogger.category` (any name that has `ilogger` is bad IMHO).

**Comment by CodeBlanch** [comment]
I really think we should shy away from trying to use the "events" spec here. It seems to be going in a much different direction. `event.domain` may even be dropped. From a comment on the spec issue, this is how we should structure these "events":

```
"event.domain": "DotNet",
"event.name": "LogEvent",
"event.data":
{
    key: "",
    categoryName: "",
    source: "",
    id: "",
    data: "",
    message: ... // etc
} 
```

**Comment by alanwest** [comment]
> event.domain is better than dotnet.ilogger.category

I agree. It's especially nasty to have something emitted with ilogger in the name from the OTLP exporter.

> I really think we should shy away from trying to use the "events" spec here. ... event.domain may even be dropped.

This was my intuition too, but I am open to trying it and continuing to solicit feedback from the community working on the event conventions. Adopting the event conventions as they are is important because the spec benefits from our participation in adopting them to drive them forward.

**Comment by vishweshbankwar** [comment]
> > event.domain is better than dotnet.ilogger.category
> 
> I agree. It's especially nasty to have something emitted with ilogger in the name from the OTLP exporter.
> 
> > I really think we should shy away from trying to use the "events" spec here. ... event.domain may even be dropped.
> 
> This was my intuition too, but I am open to trying it and continuing to solicit feedback from the community working on the event conventions. Adopting the event conventions as they are is important because the spec benefits from our participation in adopting them to drive them forward.

This is what the spec says about InstrumentationScope and it matches closely with category definition.

> For log sources which define a logger name (e.g. Java [Logger Name](https://docs.oracle.com/javase/7/docs/api/java/util/logging/Logger.html#getLogger(java.lang.String))) the Logger Name SHOULD be recorded as the Instrumentation Scope name.

This is one of the reasons I think we should not use `event.domain` for log category. Noted other reasons as well here
https://github.com/open-telemetry/opentelemetry-dotnet/pull/4925#:~:text=The%20description%20in%20the,with%20the%20Category%20Name. With InstrumentationScope name set as CategoryName we do not need to emit this attribute separately.

**Comment by alanwest** [comment]
> This is one of the reasons I think we should not use `event.domain` for log category. Noted other reasons as well [here](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4925#:~:text=For%20the%20part,the%20Category%20Name.). With InstrumentationScope name set as CategoryName we do not need to emit this attribute separately.

Good point and I agree that CategoryName could be a good candidate for instrumentation scope name. In fact, this approach would allow our handling of CategoryName be shipped stable _and_ we'd get rid of the `dotnet.ilogger.category` eyesore. For this reason, I recommend removing handling of LogRecord.CategoryName from this PR. Open a new issue to handle CategoryName as the instrumentation scope name.

This may be tricky as mentioned https://github.com/open-telemetry/opentelemetry-dotnet/pull/4925#issuecomment-1752217719, but it may be the right thing to do and we'll need to figure out a way to do it performantly.

---

Circling back to address your comment with regards to event id and name.

> These attributes have been part of the exporter since the very beginning, and not having the ability to get these attributes is disruptive for users.

No argument. However, this should not be conflated with what this PR is doing. This PR is attempting to adopt the event conventions. I am not aware of any users that have asked for that.

For now, until someone successfully champions for something different in the spec, `event.domain` is important. The only suggestion I'm left with is hard-coding it to `event.domain=ILogger`. But again, I am open to others.

Also, side note, if you read https://github.com/open-telemetry/opentelemetry-specification/issues/2994 carefully, you'll note that while the separate event.domain attribute may go away one suggestion is that its value is merged into `event.name` - e.g., `event.domain=k8s`/`event.name=some_event` becomes `event.name=k8s.some_event`. If we move forward with adopting the event conventions then as this and other issues related to the event conventions evolve, I expect a commitment from this group to keep our work aligned with the spec.

**Comment by vishweshbankwar** [comment]
> > This is one of the reasons I think we should not use `event.domain` for log category. Noted other reasons as well [here](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4925#:~:text=For%20the%20part,the%20Category%20Name.). With InstrumentationScope name set as CategoryName we do not need to emit this attribute separately.
> 
> Good point and I agree that CategoryName could be a good candidate for instrumentation scope name. In fact, this approach would allow our handling of CategoryName be shipped stable _and_ we'd get rid of the `dotnet.ilogger.category` eyesore. For this reason, I recommend removing handling of LogRecord.CategoryName from this PR. Open a new issue to handle CategoryName as the instrumentation scope name.
> 
> This may be tricky as mentioned [#4925 (comment)](https://github.com/open-telemetry/opentelemetry-dotnet/pull/4925#issuecomment-1752217719), but it may be the right thing to do and we'll need to figure out a way to do it performantly.
> 
> Circling back to address your comment with regards to event id and name.
> 
> > These attributes have been part of the exporter since the very beginning, and not having the ability to get these attributes is disruptive for users.
> 
> No argument. However, this should not be conflated with what this PR is doing. This PR is attempting to adopt the event conventions. I am not aware of any users that have asked for that.
> 
> For now, until someone successfully champions for something different in the spec, `event.domain` is important. The only suggestion I'm left with is hard-coding it to `event.domain=ILogger`. But again, I am open to others.
> 
> Also, side note, if you read [open-telemetry/opentelemetry-specification#2994](https://github.com/open-telemetry/opentelemetry-specification/issues/2994) carefully, you'll note that while the separate event.domain attribute may go away one suggestion is that its value is merged into `event.name` - e.g., `event.domain=k8s`/`event.name=some_event` becomes `event.name=k8s.some_event`. If we move forward with adopting the event conventions then as this and other issues related to the event conventions evolve, I expect a commitment from this group to keep our work aligned with the spec.

addressing https://github.com/open-telemetry/opentelemetry-dotnet/pull/4925#issuecomment-1752217719 : This is a good point. However, we already do similar thing for spans and metrics [Spans are grouped by ActivitySource and Metrics are grouped by Meter]. We have a slightly optimized way of achieving this which we could further look in to improving. But the first step should be complying with spec.


Not exporting LogRecord.CategoryName is not an option as this is also important information as per the feedback shared by users. I think we should go ahead use InstrumentationScope name for CategoryName. I am ok to try and keep that part experimental for user feedback.

Regarding the `event.domain` - I am open to using this attribute after understanding the use case for this field. How does setting `event.domain` to `ilogger` or any other value helps users?

**Comment by alanwest** [comment]
> Regarding the event.domain - I am open to using this attribute after understanding the use case for this field. How does setting event.domain to ilogger or any other value helps users?

These are questions I'm looking to you to answer. As the person recommending we use the event conventions, I'm leaning on you to describe how they're a good match for what you're trying to solve. We can't go into this ignoring the intentions of the folk who designed the conventions. We need to strive to understand their intentions and work with them to determine if we're heading down the right path. I appreciate that an issue was opened to start this discussion, but I think it's important we see it through before charging forth with whatever approach we deem convenient for our purposes.

---

If you're finding applying the event conventions is too nebulous a challenge at this point in time. I have some other ideas for moving forward.

1. Tell users to use a log processor to add the attributes.
    * This has the obvious downside that it's another component to configure and probably a bummer for performance on the logging hot path.
2. Offer experimental environment variables that enable a user to configure the attribute key themselves. Something like: 
    * `OTEL_DOTNET_EXPERIMENTAL_EXPORT_LOGRECORD_EVENT_ID_AS`
    * `OTEL_DOTNET_EXPERIMENTAL_EXPORT_LOGRECORD_EVENT_NAME_AS`

**Comment by vishweshbankwar** [comment]
> > Regarding the event.domain - I am open to using this attribute after understanding the use case for this field. How does setting event.domain to ilogger or any other value helps users?
> 
> These are questions I'm looking to you to answer. As the person recommending we use the event conventions, I'm leaning on you to describe how they're a good match for what you're trying to solve. We can't go into this ignoring the intentions of the folk who designed the conventions. We need to strive to understand their intentions and work with them to determine if we're heading down the right path. I appreciate that an issue was opened to start this discussion, but I think it's important we see it through before charging forth with whatever approach we deem convenient for our purposes.
> 
> If you're finding applying the event conventions is too nebulous a challenge at this point in time. I have some other ideas for moving forward.
> 
> 1. Tell users to use a log processor to add the attributes.
>    
>    * This has the obvious downside that it's another component to configure and probably a bummer for performance on the logging hot path.
> 2. Offer experimental environment variables that enable a user to configure the attribute key themselves. Something like:
>    
>    * `OTEL_DOTNET_EXPERIMENTAL_EXPORT_LOGRECORD_EVENT_ID_AS`
>    * `OTEL_DOTNET_EXPERIMENTAL_EXPORT_LOGRECORD_EVENT_NAME_AS`



> These are questions I'm looking to you to answer. As the person recommending we use the event conventions, I'm leaning on you to describe how they're a good match for what you're trying to solve.



> > Regarding the event.domain - I am open to using this attribute after understanding the use case for this field. How does setting event.domain to ilogger or any other value helps users?
> 
> These are questions I'm looking to you to answer. As the person recommending we use the event conventions, I'm leaning on you to describe how they're a good match for what you're trying to solve. We can't go into this ignoring the intentions of the folk who designed the conventions. We need to strive to understand their intentions and work with them to determine if we're heading down the right path. I appreciate that an issue was opened to start this discussion, but I think it's important we see it through before charging forth with whatever approach we deem convenient for our purposes.
> 
> If you're finding applying the event conventions is too nebulous a challenge at this point in time. I have some other ideas for moving forward.
> 
> 1. Tell users to use a log processor to add the attributes.
>    
>    * This has the obvious downside that it's another component to configure and probably a bummer for performance on the logging hot path.
> 2. Offer experimental environment variables that enable a user to configure the attribute key themselves. Something like:
>    
>    * `OTEL_DOTNET_EXPERIMENTAL_EXPORT_LOGRECORD_EVENT_ID_AS`
>    * `OTEL_DOTNET_EXPERIMENTAL_EXPORT_LOGRECORD_EVENT_NAME_AS`

I am not recommending to use events conventions as is. If I was, I wouldn't put `event.id` in here. But let me clarify that I am not suggesting we ignore the spec entirely. That is why working with spec in parallel is imperative here.

I see this as an opportunity to bring multiple languages in sync and not define our own conventions. It's not just .NET that needs `event.id`. Moreover, this will help us get some feedback which we can share with specification. The conventions are experimental and we are going to offer them as experimental. IMO it is ok to deviate slightly. I would like to emphasize that the plan is to not ship these stable until we resolve these differences at spec level. 

And to add, we have not stopped for spec to ship stable things in the past so, I would like to understand the rationale behind blocking changes that depends on experimental spec and experimental feature flag. 

Additionally, I don't think we should be doing any workarounds. Setting experimental feature flag itself is workaround we are asking users to do on a stable library.

**Comment by vishweshbankwar** [comment]
In favor of making progress and unblocking this, I have updated the attributes to be `logrecord.event.id` and `logrecord.event.name`. Regarding the category name, have opened https://github.com/open-telemetry/opentelemetry-dotnet/pull/4941 to export that as InstrumentationScope name under ScopeLogs.

@open-telemetry/dotnet-approvers @open-telemetry/dotnet-maintainers Please review.

**Review by utpilla — APPROVED** [review]

**Review by rajkumar-rangaraj — APPROVED** [review]

**Review by alanwest — APPROVED** [review]

**Pull request merged**
